rss/index.py

```diff
@@ -18,70 +22,189 @@ def perform_search(archives: str, date_time: datetime.datetime) -> EPrintSet:
 
-        # TODO - Compose a query for possibly multiple archives
-        # TODO - Compose a query for the last day's papers from the specified archive
-        q = Q("match_all")
-
-        # TODO - Add all requested categories to the eprints struct
-
-        # Perform the search and create an EPrint object for every hit that is found
-        response = Search(index='arxiv').using(es).query(q).execute()
-        for hit in response:
-            # Collect top-level properties from the hit
-            archive = hit["primary_classification"]["archive"]
-            arxiv_id = archive["id"]
-            archive_name = archive["name"]
-            paper_id = hit['paper_id']
-            title = hit['title']
-            abstract = hit['abstract']
-            submitted_date = hit['submitted_date']
-            updated_date = hit['updated_date']
-            comments = hit['comments']
-            journal_ref = hit['journal_ref']
-            doi = hit['doi']
-
-            # Copy the hit data for authors into the EPrint
-            authors = list()
-            for hit_author in hit['authors']:
-                last_name = hit_author['last_name']
-                full_name = hit_author['full_name']
-                initials = hit_author['initials']
-                author = Author(last_name, full_name, initials)
-                authors.append(author)
-
-            # Copy the hit data for categories into the EPrint
-            category = hit['primary_classification'].to_dict()['category']
-            primary_category = Category(category['name'], category['id'])
-            secondary_categories = list()
-            for classification in hit['secondary_classification']:
-                category = classification['category'].to_dict()
-                secondary_category = Category(category['name'], category['id'])
-                secondary_categories.append(secondary_category)
-
-            # Create the EPrint and add it to the collection to be returned
-            eprint = EPrint(arxiv_id, archive_name, paper_id, title, abstract,
-                            submitted_date, updated_date, comments, journal_ref, doi,
-                            authors, primary_category, secondary_categories)
-            eprints.append(eprint)
+        # Perform the search, filtering to only return the last day's papers
+        start_date = (date_time - datetime.timedelta(days=days)).strftime('%Y-%m-%d')
+        end_date = date_time.strftime('%Y-%m-%d')
+        response = Search(index='arxiv')\
+            .using(es) \
+            .query(q) \
+            .filter("range", submitted_date={'gte': start_date, 'lte': end_date, 'format': 'date'}) \
+            .execute()
+        return status.HTTP_200_OK, response
+        # .filter("range", submitted_date={'gte': 'now-1d/d', 'lt': 'now/d'})
+
+    except ElasticsearchException:
+        return status.HTTP_500_INTERNAL_SERVER_ERROR, list()
```

Comment:
Same thing down here -- the ``rss.index`` module shouldn't know anything/care about the status codes we are returning. If a problem is encountered, you could raise an exception to indicate to the controller that something went wrong. The controller would then decide to return 500 (or something else).

Here's a (maybe) simple example to illustrate what I mean:

```python

def do_service_things(foo):    # Here is a function in our service integration module.
    result = _get_a(foo)  # One of the goals is to screen off whatever shenanigans happen herein.
    if result == 1:
        return 'hurray'
    raise SomethingIsFishy(f'Got a {result}')


def handle_a_request(params):    # Here is our controller function.
    foo = params.get('bar')     # The controller is responsible for figuring out what is relevant in the request.
    try:
        result = do_service_things(foo)  # The service module does its thing, or panics.
    except SomethingIsFishy as e:
        if _is_expected(e):    # Just because the service module panics doesn't mean we have an error.
            return status.HTTP_200_OK, {'alles': 'ist gut'}, {}
        raise InternalServerError('Things are really squirrely right now') from e  # Chaining is nice.
    return status.HTTP_200_OK, {'result': result}, {}

```

Reply:
Thanks for the example!